Realm/RLMMigration.h

```diff
@@ -75,15 +75,17 @@ typedef void (^RLMObjectMigrationBlock)(RLMObject * __nullable oldObject, RLMObj
 
 /**
  Creates and returns an `RLMObject` instance of type `className` in the Realm being migrated.
+ 
+ The `value` argument is used to populate the object. It can be a key-value coding compliant object, an array or 
+ dictionary returned from the methods in `NSJSONSerialization`, or an `NSArray` containing one element for each managed
+ property. An exception will be thrown if any required properties are not present and those properties were not defined
+ with default values. Do not pass in a `RLMLinkingObjects` instance, either by itself or as a member of a collection.
+
+ When passing in an `NSArray` as the `value` argument, all properties must be present, valid and in the same order as
```

Comment:
we interchangeably use "`NSArray`" or "array" in this comment. Let's stick with "array".